python/pyarrow/scalar.pxi

```diff
@@ -687,6 +687,44 @@ cdef class DictionaryScalar(Scalar):
     Concrete class for dictionary-encoded scalars.
     """
 
+    def __init__(self, index, dictionary, type):
```

Comment:
Maybe we can put this in some `from_..` class method? (to keep it consistent with the other scalars that `__init__` raises)

Reply:
(and can you also add some tests for this construction method?)